### ISSUE TITLE
[java] JUnit4TestShouldUseBeforeAnnotation false positive when overriding setUp

### ISSUE THREAD

**Issue by barriosnahuel** [pmd/pmd#1592]
**Affects PMD Version:** v6.8.0

**Rule:** `JUnit4TestShouldUseBeforeAnnotation`, `JUnit4TestShouldUseAfterAnnotation`

**Description:** When overriding a `setUp` method from a superclass which already has the `@Before` annotation the rule throws a false positive.

**Code Sample demonstrating the issue:**

```
public class BaseTest {
    @Before
    @CallSuper
    public void setUp() {
        // A setup code
    }
    @After
    @CallSuper
    public void tearDown() {
        // A tear down code
    }
}

public class AReallyCoolFeatureTest extends BaseTest {
    @Override
    public void setUp() {
        super.setUp();
    }

    @Override
    public void tearDown() {
        super.tearDown();
    }
}
```

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
Given a complete auxclasspath, checking for annotations on overridden methods is plausible. However, that is currently not possible from an XPath rule... To perform this fix we would have to rewrite the rule in Java.

An alternative, would be simply to ignore all methods annotated with `@Override` assuming the base class is properly annotated; which may swing this from false positive to false negative...

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
